addon/services/permissions.ts

```diff
@@ -33,37 +41,29 @@ export default class PermissionsService extends Service {
     sendEvent(this, EVENTS.ROUTE_PERMISSIONS_CHANGED);
   }
 
-  cacheInitialTransition() {
-    addListener(
-      this.routerService,
-      'routeWillChange',
-      this,
-      (transition) => {
-        this.initialTransition = transition;
-      },
-      true
-    );
+  cacheInitialTransition(): void {
+    this.routerService.on('routeWillChange', (transition: Transition) => {
+      this.initialTransition = transition;
+    });
 
-    addListener(
-      this.routerService,
-      'routeDidChange',
-      this,
-      () => {
-        this.initialTransition = null;
-      },
-      true
-    );
+    this.routerService.on('routeDidChange', () => {
+      this.initialTransition = null;
+    });
   }
 
-  hasPermissions(...args) {
-    const permissions = Array.isArray(args[0]) ? args[0] : args;
+  hasPermissions(permissions: Permission[]): boolean;
```

Comment:
[Function overloads ](https://www.typescriptlang.org/docs/handbook/functions.html#overloads)are needed here because it's possible to invoke it in 2 different ways (with string params or an array of strings).

I think it might be a good idea to remove the array support in the future since that would simplify the implementation (and the type) and the churn on the invocation side can be limited by using spread syntax on the array.